dz-9/src/main/java/org/example/Woman.java

```diff
@@ -0,0 +1,34 @@
+package org.example;
+
+public class Woman extends Person {
+
+    Woman (String firstName, String lastName, Integer age) {
+        super(firstName, lastName, age);
+
+    }
+    @Override
+    boolean isRetired() {
+        if (age>=60) {
+            return true;
+        }else{
+            return false;
+        }
+    }
+
+    @Override
+    void registerPartnership(Person man) {
+        this.setLastName(man.getLastName());
+        this.partner = man;
```

Comment:
You do not check that in argument `man` actually instance of class `Man`/ Also you need to set partner for partner in the same method